patches/api/0417-Add-PlayerPreRespawnLocationEvent.patch

```diff
@@ -0,0 +1,80 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cryptite <cryptite@gmail.com>
+Date: Mon, 1 May 2023 15:23:34 -0500
+Subject: [PATCH] Add PlayerPreRespawnLocationEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerPreRespawnLocationEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerPreRespawnLocationEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..cf81752c60fb1cd5a649c50acea759036ba9d094
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerPreRespawnLocationEvent.java
+@@ -0,0 +1,68 @@
++package io.papermc.paper.event.player;
++
++import org.apache.commons.lang3.Validate;
++import org.bukkit.Location;
++import org.bukkit.entity.Player;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.bukkit.event.player.PlayerRespawnEvent;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Called when a respawn event tries to determine the location of a respawn. This is called before {@link PlayerRespawnEvent}
++ */
++public class PlayerPreRespawnLocationEvent extends PlayerEvent {
++    private static final HandlerList handlers = new HandlerList();
```

Comment:
Use the standard for static final fields, aka all caps (e.g. `HANDLER_LIST`). Ignore bukkit's dum way of just doing `handlers`.